nekoyume/Assets/_Scripts/State/Modifiers/AvatarInventoryTradableItemRemover.cs

```diff
@@ -1,48 +1,34 @@
 ﻿using System;
 using System.Collections.Generic;
-using Nekoyume.JsonConvertibles;
+using System.Linq;
 using Nekoyume.Model.State;
-using UnityEngine;
 
 namespace Nekoyume.State.Modifiers
 {
     [Serializable]
     public class AvatarInventoryTradableItemRemover : AvatarStateModifier
     {
-        [Serializable]
-        public class InnerDictionary : JsonConvertibleDictionary<JsonConvertibleGuid, int>
+        private class InnerModel
         {
-        }
-
-        [SerializeField]
-        private InnerDictionary innerDictionary;
+            public Guid Guid { get; }
+            public long RequiredBlockIndex { get; }
+            public int Count { get; set; }
 
-        public override bool IsEmpty => innerDictionary.Value.Count == 0;
-
-        public AvatarInventoryTradableItemRemover(Guid tradableId, int count = 1)
-        {
-            if (count is 0)
+            public InnerModel(Guid guid, long requiredBlockIndex, int count)
             {
-                innerDictionary = new InnerDictionary();
-                return;
+                Guid = guid;
+                RequiredBlockIndex = requiredBlockIndex;
+                Count = count;
             }
-
-            innerDictionary = new InnerDictionary();
-            innerDictionary.Value.Add(new JsonConvertibleGuid(tradableId), count);
         }
 
-        public AvatarInventoryTradableItemRemover(Dictionary<Guid, int> dictionary)
-        {
-            innerDictionary = new InnerDictionary();
-            foreach (var pair in dictionary)
-            {
-                if (pair.Value is 0)
-                {
-                    continue;
-                }
+        private List<InnerModel> _items = new List<InnerModel>();
```

Comment:
How about mapping into dictionary like this?
```
private class InnerModel
{
            public long RequiredBlockIndex { get; }
            public int Count { get; set; }
}

private Dictionary<JsonConvertibleGuid, InnerModel> _items = new List<InnerModel>();
```

Reply:
Please review one more time. :pray: